sandbox/libs/authn/src/main/java/org/opensearch/authn/internal/InternalSubject.java

```diff
@@ -64,7 +65,48 @@ public String toString() {
      */
     public void login(AuthenticationToken authenticationToken) {
         org.apache.shiro.authc.AuthenticationToken authToken = AuthenticationTokenHandler.extractShiroAuthToken(authenticationToken);
+
+        // If already authenticated, do not check login info again
+        /*
+            TODO: understand potential repercussions in following situations:
+             1. How to handle this in password change situations
+             2. Can two subjects in same environment have same principal name? if so the following check is invalid
+         */
+        if (this.isAuthenticated() && this.getPrincipal().getName().equals(authToken.getPrincipal())) {
```

Comment:
Principal identifiers are unique within the same realm.

What is the purpose of the second part of this check?

Reply:
The test would fail when:
1. Admin sends a request and is valid. Subject's principal is set to `admin` and isAuthenticated is set to true
2. Next, a dummy user with invalid credentials tries to login. Since isAuthenticated is true, this would assume a valid user and skip login without second condition. With second condition we match `dummy` and `admin` and see that they don't match so incoming request should go through authentication workflow

Although I'm not 100% sure about this behavior. How should `Subject` be treated for each incoming request?